src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -105,7 +107,37 @@ export default class DefaultMessagingSession implements MessagingSession {
   }
 
   private async startConnecting(reconnecting: boolean): Promise<void> {
-    const signedUrl = await this.prepareWebSocketUrl();
+    let endpointUrl = this.configuration.endpointUrl;
+    // reconnect needs to re-resolve endpoint url, which will also refresh credentials on client if they are expired
+    if (reconnecting || endpointUrl === undefined) {
+      try {
+        if (this.configuration.chimeClient.getMessagingSessionEndpoint instanceof Function) {
+          endpointUrl = (
+            await this.configuration.chimeClient.getMessagingSessionEndpoint()
+          ).Endpoint.Url;
+        } else {
+          endpointUrl = (
+            await this.configuration.chimeClient.send(new GetMessagingSessionEndpointCommand({}))
+          ).Endpoint.Url;
+        }
+        this.logger.debug(`Messaging endpoint resolved to: ${endpointUrl}`);
+      } catch (e) {
+        this.logger.error("DPW " + e)
```

Comment:
Is this intended?

Reply:
this diff is old I believe (was having some struggles commiting changes) please see latest